radis/api/kuruczapi.py

```diff
@@ -0,0 +1,490 @@
+                dict_entries.update(
+                    {
+                        "info": info,
+                        "wavenumber_min": self.wmin,
+                        "wavenumber_max": self.wmax,
+                    }
+                )
+        # else:
+        #     raise Exception("Kurucz database can't be registered until the correct url to use is known")
+
+        if get_pf_files:
+            # files.append(self.pf_path)
+            # urls.append(self.pf_url)
+            dict_entries.update(
+                {
+                    "parfunc": self.pf_path,
+                    "parfuncfmt": "kurucz",
+                }
+            )
+
+        dict_entries.update(
+            {
+                "parfuncfmt": "kurucz",
+            }
+        )
+
+        try:
+            super().register(dict_entries)
+        except DatabaseAlreadyExists as e:
+            raise Exception(
+                'If you want RADIS to overwrite the existing entry for a registered databank, set the config option "ALLOW_OVERWRITE" to True.'
+            ) from e
+
+
+def read_kurucz(kuruczf, preserve_orig_levels=False):
+    """
+    Parse a Kurucz linelist, process its columns as required, and return as a Pandas DataFrame
+
+    Inspired by: https://github.com/rasmus98/NLTE-Helium/blob/fc6161a30ebecfcf59f36386b1dc7f02ff749905/Grotrian_diagram_Helium.ipynb cell 2. Also see:
+    - https://github.com/DBerke/varconlib/blob/e57250ca359026ae8b8059dae179fb0ad9625aa2/varconlib/scripts/select_line_pairs.py#L711
+    - https://github.com/followthesheep/specutils/blob/e2873719f3820e83ab29c79062d7a59a7664fa2f/specutils/read_kurucz_linelist.py
+    - https://github.com/ajwheeler/Korg.jl/blob/53420d38c23e21a7fe202b8680bdb201c9a62a2a/src/linelist.jl#L153
+
+    Parameters
+    ----------
+    kuruczf: file path
+    preserve_orig_levels: (boolean)
+        whether to preserve the original columns pertaining to the levels prior to transforming them, with '_orig' appended to their name; note the columns are still reversed so that the database index increases in wavenumber rather than wavelength
+
+    Returns:
+        Pandas DataFrame containing the required columns for the Kurucz linelist database
+    """
+    colspecs = (
+        (0, 11),
+        (11, 18),
+        (18, 24),
+        (24, 36),
+        (36, 41),
+        (42, 52),
+        (52, 64),
+        (64, 69),
+        (70, 80),
+        (80, 86),
+        (86, 92),
+        (92, 98),
+        # (98,102),(102,104),(104,106),
+        (106, 109),
+    )
+    # (109,115),(115,118),(118,124),(124,129),(129,134),(135,136),(136,137),(138,139),(139,140),(140,141),(141,144),(144,149),(149,154),(154,160))
+    dtypes = (
+        np.float64,
+        np.float64,
+        "object",  # make species string rather than float for easier parsing
+        np.float64,
+        np.float64,
+        "object",
+        np.float64,
+        np.float64,
+        "object",
```

Comment:
Is it not better to use string rather than objects?

Reply:
I changed the dtype for the three string columns from 'string' to 'object' due to an error appearing in [Python 3.11 tests](https://app.travis-ci.com/github/radis/radis/jobs/624031722) `AttributeError: 'StringDtype' object has no attribute 'itemsize'`

Reply:
Seems to be related to [this](https://github.com/pandas-dev/pandas/issues/31199)